### DISCUSSION TITLE
Retrieve a micro in local and update it with already made changes

### DISCUSSION THREAD

**Discussion by aaronted009** [deta/cloud-docs#526] [Help] [answered]
Hello.
I made a micro server and deployed it from my laptop. The micro is up and running.
It happened that I started working then on a work station where I retrieved the project with already lot of updates.
I started configuring the work station with deta. 

Installing it as [described](https://docs.deta.sh/docs/micros/getting_started/#configuring--installing-the-deta-cli).
Logged in : `deta login`
I cloned it : `deta clone --name <name> --project <project>`. 
But this cloned micro is just creating a whole folder of the previously deployed project into the current updated one.

What I'm trying to achieve is to deploy the actual updated project to the current micro server already running rather than creating a new one.
Any idea on how I can achieve that ?

I also tried :
`deta pull` among other commands most resulting in telling there's no micro initialized. Does this mean I necessarily have to create a new micro rather than using the one already deployed ?

**Comment by etrotta** [comment, marked as answer]
turn on `Show hidden files` or whatever it is called in your OS, then move the `.deta` folder from the folder generated by clone to the updated one and try deploying
if things breaks you can try deleting the state file in that folder and deploying it again, or just deleting that micro and creating a new one

**Reply by rohanshiva** [reply]
@aaronted009 can you please try this?

Run the clone command in a different place, copy `.deta` folder without the state file to the folder containing the updated src code and run `deta deploy` in the folder containing the updated src code.

Please let me know if any of this is unclear.

**Reply by aaronted009** [reply]
Hello, thanks for your answers.
I've tried what you suggested :
`.deta` folder moved into the updated project then `deta deploy`:
```
Deploying...
Successfully deployed changes
Updating dependencies...
Error: failed to update dependencies: Program already has dependency fastapi==0.79.1 installed
```
I think it already made the updates I want. But because of the error, I tried deleting the state file and deploy it again.
I got the same message :
```
Deploying...
Successfully deployed changes
Updating dependencies...
Error: failed to update dependencies: Program already has dependency fastapi==0.79.1 installed
```
I guess I can leave the state file in the folder right ?
By now changes have been deployed, so well that I'm now getting a `Bad Gateway`...
Can you help with this ? In the logs, I see there's a `ModuleNotFoundError` : 
```
...
    import itsdangerous
ModuleNotFoundError: No module named 'itsdangerous'
```
I tried `deta update -e .env` and even installing again the package (which already has been) but I have the same error in the logs.
I think dependencies are not updated as `deta details` is not showing all current dependencies. 
`deta deploy` keeps giving :
```
Updating dependencies...
Error: failed to update dependencies: Program already has dependency fastapi==0.79.1 installed
```

**Reply by aaronted009** [reply]
Well, it's about duplicates in my `requirements.txt` file. Deleted the duplicates and it's good now.
My first issue is solved anyway.
Thanks @etrotta @rohanshiva
